src/network/networkd-link.c

```diff
@@ -1829,6 +1872,14 @@ void link_drop(Link *link) {
 
         log_link_debug(link, "Link removed");
 
+        if (link->alternative_names) {
+                char **n;
+
+                STRV_FOREACH(n, link->alternative_names)
+                        hashmap_remove_value(link->manager->links_by_name, *n, link);
+        }
+        hashmap_remove_value(link->manager->links_by_name, link->ifname, link);
```

Comment:
please move the name unregistration into a function of its own, too, also covering both the alternative names and the main name